src/Persisters/DocumentPersister.php

```diff
@@ -994,7 +994,7 @@ public function prepareFieldName(string $fieldName): string
      */
     public function addDiscriminatorToPreparedQuery(array $preparedQuery): array
     {
-        if (isset($preparedQuery[$this->class->discriminatorField]) || $this->class->discriminatorField === null) {
+        if (isset($preparedQuery[$this->class->discriminatorField ?? '']) || $this->class->discriminatorField === null) {
```

Comment:
```suggestion
        if ($this->class->discriminatorField === null || isset($preparedQuery[$this->class->discriminatorField])) {
```

Reversing the condition order is simpler.

Reply:
Smart, thank you.